sample-app/src/main/java/org/smartregister/pnc/sample/activity/PncFormActivity.java

```diff
@@ -1,22 +1,89 @@
-package org.smartregister.pnc.sample.activity;
-
-
-import com.vijay.jsonwizard.constants.JsonFormConstants;
-
-import org.smartregister.pnc.activity.BasePncFormActivity;
-import org.smartregister.pnc.sample.R;
-import org.smartregister.pnc.sample.fragment.PncFormFragment;
-
-
-public class PncFormActivity extends BasePncFormActivity {
-
-    @Override
-    public void initializeFormFragment() {
-        initializeFormFragmentCore();
-    }
-
-    protected void initializeFormFragmentCore() {
-        PncFormFragment pncFormFragment = PncFormFragment.getFormFragment(JsonFormConstants.FIRST_STEP_NAME);
-        getSupportFragmentManager().beginTransaction().add(R.id.container, pncFormFragment).commit();
-    }
-}
+package org.smartregister.pnc.sample.activity;
+
+
+import android.support.annotation.NonNull;
+
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.pnc.activity.BasePncFormActivity;
+import org.smartregister.pnc.config.PncChildStatusRepeatingGroupGenerator;
+import org.smartregister.pnc.sample.R;
+import org.smartregister.pnc.sample.fragment.PncFormFragment;
+import org.smartregister.pnc.utils.PncConstants;
+import org.smartregister.pnc.utils.PncDbConstants;
+import org.smartregister.repository.BaseRepository;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import timber.log.Timber;
+
+
+public class PncFormActivity extends BasePncFormActivity {
+
+    @Override
+    public void initializeFormFragment() {
+        initializeFormFragmentCore();
+    }
+
+    protected void initializeFormFragmentCore() {
+        PncFormFragment pncFormFragment = PncFormFragment.getFormFragment(JsonFormConstants.FIRST_STEP_NAME);
+        getSupportFragmentManager().beginTransaction().add(R.id.container, pncFormFragment).commit();
+    }
+
+    @Override
+    public void init(String json) {
+        try {
+            JSONObject jsonObject = new JSONObject(json);
+
+            generateRepeatingGrpFields(jsonObject);

```

Comment:
this should not be called form here ... it should be called from starting activity